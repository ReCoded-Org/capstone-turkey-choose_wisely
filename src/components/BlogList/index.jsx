import React, { useEffect, useState } from "react";
import BlogDetails from "../BlogDetails";
import { useTranslation } from "react-i18next";
import { Container, Row, Col } from "react-bootstrap";
import { LANGUAGES } from "./../../utilities/variables";

const BlogList = () => {
  const [posts, setPosts] = useState([]);
  const { t, i18n } = useTranslation();

  useEffect(() => {
    const fetchPostData = async () => {
      try {
        const [language] = LANGUAGES.filter(
          (language) => language.code === i18n.language
        );
        const posts = await fetch(
          `https://public-api.wordpress.com/wp/v2/sites/choosewisely834444303.wordpress.com/posts?tags=${language.key}`
        );
        const data = await posts.json();
        setPosts(data);
      } catch (error) {
        console.log(error);
      }
    };
    fetchPostData();

    setTimeout(() => {
      const aNods = document.querySelectorAll(".more-link");
      const array = Array.from(aNods);
      array.forEach((element) => {
        element.remove();
      });
    }, 900);
    // eslint-disable-next-line
  }, [i18n?.language]);

  const categories = [
    { key: 1342, value: "Education" },
    { key: 103, value: "News" },
    { key: 405, value: "Campus" },
    { key: 3072, value: "IT" },
  ];

  return (
    <div>
      <Container>
        <div>
          <Row>
            <Col lg={12}>
              <h1 className="header">{t("blogs.header")}</h1>
            </Col>

            <Col lg={6} md={12}>
              <p className="blogsParagraph">{t("blogs.title")}</p>
            </Col>
          </Row>
        </div>

        {posts.map((post) => {
          const categoryObject = categories.find(
            (el) => el.key === post.categories[0]
          );
          post.category = categoryObject.value;
          return <BlogDetails post={post} />;
        })}
      </Container>
    </div>
  );
};

export default BlogList;
