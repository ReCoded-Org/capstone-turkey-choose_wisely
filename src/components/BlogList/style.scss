.HeaderAndTitleSection {
  width: 6.875rem;
  height: 7.5rem;
  padding-top: 1.875rem;

  padding-left: 0.25rem;
}
