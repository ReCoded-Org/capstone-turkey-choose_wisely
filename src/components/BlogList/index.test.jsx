import React from "react";
import BlogList from "./index";
import renderer from "react-test-renderer";

jest.mock("react-i18next", () => ({
  useTranslation: () => ({ t: (key) => key }),
}));

const blog = {
  header: "Latest Blog Posts",
  title:
    "Home is behind, the world ahead and there are many paths to tread through shadows to the edge.",
};
test("Snapshop test of BlogList component", () => {
  const map = renderer.create(<BlogList {...blog} />);
  let tree = map.toJSON();

  expect(tree).toMatchSnapshot();
});
