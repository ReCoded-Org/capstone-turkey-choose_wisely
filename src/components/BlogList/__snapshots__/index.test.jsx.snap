// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshop test of BlogList component 1`] = `
<div>
  <div
    className="container"
  >
    <div>
      <div
        className="row"
      >
        <div
          className="col-lg-12"
        >
          <h1
            className="header"
          >
            blogs.header
          </h1>
        </div>
        <div
          className="col-lg-6 col-md-12"
        >
          <p
            className="blogsParagraph"
          >
            blogs.title
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
